scripts/scripts.js

```diff
@@ -1272,18 +1329,35 @@ if (params.get('performance')) {
  * Registers the 'convert' function to `sampleRUM` which sends
  * variant and convert events upon conversion.
  */
-sampleRUM.drain('convert', (elements, cevent, cvalue) => {
+sampleRUM.drain('convert', (elements, cevent, cvalue, extraData) => {
   // if elements is an array or nodelist, register a conversion event for each element
   if (Array.isArray(elements) || elements instanceof NodeList) {
     elements.forEach((element) => {
       // if the element is a form, register a submit event
       if (element.tagName === 'FORM') {
-        element.addEventListener('submit', () => {
-          sampleRUM.convert(element, cevent, cvalue);
+        element.addEventListener('submit', () => {          
+          const evtDataLayer = {
+            event: "Form Complete",
+            forms: {
+               formsComplete: 1,
+               formName: cevent,
+               formId: element.id,
+               formsType: "" 
+            }
+          };
+          sampleRUM.convert(element, cevent, cvalue, evtDataLayer);
         });
       } else {
-        element.addEventListener('click', () => {
-          sampleRUM.convert(element, cevent, cvalue);
+        element.addEventListener('click', () => {          
+          const evtDataLayer = {
+            event: "Link Click",
+            eventData: {
+              linkName: cevent,
+              linkText: element.innerHTML,
+              linkHref: element.href              
+            }
+          };
+          sampleRUM.convert(element, cevent, cvalue, evtDataLayer);
```

Comment:
We should revert this bit. The `evtDataLayer` stuff is application-level and customer-dependent. I see the `sampleRUM.convert` as library code that should not have customer specifics.

Reply:
Done